bgpd/bgp_io.c

```diff
@@ -387,8 +387,10 @@ static uint16_t bgp_write(struct peer *peer)
 		int writenum;
 		do {
 			writenum = stream_get_endp(s) - stream_get_getp(s);
+			if (peer->bgp)
+				vrf_switch_to_netns(peer->bgp->vrf);
```

Comment:
This is not thread safe and introduces a race condition. There are statics at the bottom of this call chain. Please run your tests under Helgrind or clang's ThreadSanitizer to catch this in the future.